OpenRA.Mods.Common/Traits/Render/WithAttackAnimation.cs

```diff
@@ -51,36 +45,30 @@ public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
 
 	public class WithAttackAnimation : ConditionalTrait<WithAttackAnimationInfo>, ITick, INotifyAttack
 	{
-		readonly AttackBase attack;
 		readonly Armament armament;
 		readonly WithSpriteBody wsb;
-		readonly bool noAimOrReloadAnim;
 
 		int tick;
-		bool attackAnimPlaying;
 
 		public WithAttackAnimation(ActorInitializer init, WithAttackAnimationInfo info)
 			: base(info)
 		{
-			attack = init.Self.Trait<AttackBase>();
 			armament = init.Self.TraitsImplementing<Armament>()
 				.Single(a => a.Info.Name == Info.Armament);
 			wsb = init.Self.TraitsImplementing<WithSpriteBody>().First(w => w.Info.Name == Info.Body);
-
-			noAimOrReloadAnim = string.IsNullOrEmpty(Info.AimSequence) && string.IsNullOrEmpty(Info.ReloadPrefix);
 		}
 
 		void PlayAttackAnimation(Actor self)
 		{
-			if (!IsTraitDisabled && !wsb.IsTraitDisabled && !string.IsNullOrEmpty(Info.AttackSequence))
-			{
-				attackAnimPlaying = true;
-				wsb.PlayCustomAnimation(self, Info.AttackSequence, () => attackAnimPlaying = false);
-			}
+			if (!IsTraitDisabled && !wsb.IsTraitDisabled && !string.IsNullOrEmpty(Info.Sequence))
+				wsb.PlayCustomAnimation(self, Info.Sequence);
 		}
 
 		void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barrel)
 		{
+			if (a != armament)
```

Comment:
This can be merged into the check below: `if (a == armament && Info.DelayRelativeTo == AttackDelayType.Attack)`